### PR TITLE
Fix redundant resources

### LINKED ISSUE #67: update resource specification to avoid redundancy
Currently threads for each rule are specified in the rule itself and also in the `cluster.yaml` files. This is an issue for other resources as well, but to lesser extent.  

- `--cpus-per-task={threads}`   instead of `--cpus-per-task={cluster.cpus}` in Snakefiles
- pare down cluster.yaml files
- specify tmpdir in sbatch scripts
- add `resources` to relevant rules

### PR DESCRIPTION
**Pull request by juniper-lake**
- Removed all `cluster.yaml` files by automatically assigning CPUs based on {threads} and moving additional resources requirements of DeepVariant to the rules.
- Added example batch scripts for lsf
- Added profiles for slurm, sge, and lsf to define default-resources
- Moved tmpdir specification from config to profiles and confirmed that both TMPDIR and SINGULARITY_TMPDIR are correctly assigned

resolves #67

### CODE REVIEW

File: profiles/lsf/config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+conda-frontend: mamba
+latency-wait: 120
+cluster: bsub -cwd 
+              -q {resources.partition}
+              -n {threads}
+              -o {resources.out}
+              -e {resources.err} {resources.extra}
+default-resources: 
+  - partition='default'
+  - tmpdir=system_tmpdir
```
Comment (williamrowell) [inline profiles/lsf/config.yaml:16]: Does this get passed through correctly to containers?

Comment (juniper-lake) [reply profiles/lsf/config.yaml:16]: Also, the use of tmpdir in --default-resources requires a newer version of snakemake (>=6.5.2), but I tested on a more recent version so the environment.yaml was updated to reflect the testing version (>=6.15).

**Review by williamrowell — CHANGES_REQUESTED** [review]
Looks good.  It's all tested?

One tiny change and a (non-binding) suggestion to add USAGE lines to the shell scripts.

**Review by williamrowell — APPROVED** [review]

**Comment by juniper-lake** [comment]
Tested slurm and local (80 cores), completed all steps of workflow

**Review by williamrowell — APPROVED** [review]
This looks great.  Thanks for adding the USAGE statements.

**Pull request merged**